### PR TITLE
Fix having an extra 'default' section when the last row is a leaf

### PR DESCRIPTION
**Pull request by mgeplf**
* Rename Section -> to DataBlockSection so it's not confused with
  neurom.Section

### CODE REVIEW

File: neurom/core/dataformat.py [hunk 1]
```diff
@@ -29,9 +29,13 @@
 '''Data format definitions'''
 
 
+_COL_COUNT = 7
```
Comment (mgeplf) [inline neurom/core/dataformat.py]: cleanup

File: neurom/io/datawrapper.py [hunk 2]
```diff
@@ -168,31 +177,30 @@ def _extract_sections(data_block):
             curr_section.ids.append(row_id)
 
         if row_id in sec_end_pts:
-            parent_section[curr_section.ids[-1]] = len(_sections) - 1
-            _sections.append(Section())
-            curr_section = _sections[-1]
+            parent_section[curr_section.ids[-1]] = len(sections) - 1
             # Parent-child discontinuity section
             if gap:
+                sections.append(DataBlockSection())
```
Comment (mgeplf) [inline neurom/io/datawrapper.py]: this is where the problem is: an extra `DataBlockSection` was being created unnecessarily for the last row if it was a leaf

File: neurom/io/datawrapper.py [hunk 3]
```diff
@@ -168,31 +177,30 @@ def _extract_sections(data_block):
...
-
-
-COL_COUNT = 7
-_, _, _, _, TYPE, ID, PARENT = range(COL_COUNT)
+    # TODO find a way to remove empty sections.  Currently they are
+    # required to maintain tree integrity.
+    return sections
 
 
 class BlockNeuronBuilder(object):
```
Comment (mgeplf) [inline neurom/io/datawrapper.py]: everything from here on is just cleanup

File: neurom/io/datawrapper.py [hunk 4]
```diff
@@ -168,31 +177,30 @@ def _extract_sections(data_block):
...
             if gap:
+                sections.append(DataBlockSection())
+                curr_section = sections[-1]
                 curr_section.ids.extend((parent_id, row_id))
                 curr_section.ntype = int(row[COLS.TYPE])
-                _gap_sections.add(len(_sections) - 2)
+                gap_sections.add(len(sections) - 2)
+            elif row_id != len(data_block) - 1:
+                # avoid creating an extra DataBlockSection for last row if it's a leaf
+                sections.append(DataBlockSection())
```
Comment (arsenius7) [inline neurom/io/datawrapper.py]: Does it worth making these two lines a helper method or removing curr_section altogether?

Comment (mgeplf) [inline neurom/io/datawrapper.py]: I thought about that when writing it.  I think I'm going to rewrite some of this routine and its dependencies in a more 'numpy' way, so I'll do it then, if that's ok.

**Pull request merged**